services/app/src/components/header/CreateNewRoomBtn.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useNavigate } from 'react-router';
+import { nanoid } from 'nanoid';
+import { ROOM_ID_LENGTH } from '../../constants/appConstants';
+import Button from '../button/Button';
+import RefreshIcon from '../../static/svg/RefreshIcon.svg';
+import { buildPathFromTemplate, ROUTES } from '../../pages/routes';
+import { useUserContext } from '../../contexts/UserContext';
+import { useWsContext } from '../../contexts/ws-context';
+import { buildPlayerConnectedEventMessage } from '../../api/playerApi';
+
+const CreateNewRoomBtn = () => {
+  const navigate = useNavigate();
+  const userContext = useUserContext();
+  const { ws, openWsConnection } = useWsContext();
+
+  const createNewRoom = () => {
+    ws.close();
+    const newRoomId = nanoid(ROOM_ID_LENGTH);
+    navigate(
+      buildPathFromTemplate(ROUTES.game, {id: newRoomId}),
+      { replace: true }
+    );
+    
+    openWsConnection({
```

Comment:
Is it possible to have a logic for opening a new connection in a single place? I.e. in the game room only or wherever it is placed at the moment?